components/Navbar.tsx

```diff
@@ -1,149 +1,325 @@
+            {!!props.session && <DesktopNav />}
+          </Flex>
+        </Flex>
+
+        <Stack
+          flex={{ base: 1, md: 0 }}
+          justify={"flex-end"}
+          direction={"row"}
+          spacing={6}
+        >
+          {!props.session && (
             <Button
-              variant="outlined"
-              color="info"
-              sx={{ backgroundColor: "white" }}
+              as={"a"}
+              fontSize={"sm"}
+              fontWeight={400}
+              variant={"link"}
               onClick={() => navigateToAuthPage()}
+              href={"#"}
             >
-              login
+              Sign In
             </Button>
           )}
-        </Toolbar>
-      </AppBar>
-      <Box component="nav">
-        <Drawer
-          container={undefined}
-          variant="temporary"
-          open={mobileOpen}
-          onClose={handleDrawerToggle}
-          ModalProps={{
-            keepMounted: true, // Better open performance on mobile.
-          }}
-          sx={{
-            display: { xs: "block", sm: "none" },
-            "& .MuiDrawer-paper": {
-              boxSizing: "border-box",
-              width: drawerWidth,
-            },
-          }}
-        >
-          {drawer}
-        </Drawer>
-      </Box>
+          {!!props.session && (
+            <Menu>
+              <MenuButton
+                as={Button}
+                rounded={"full"}
+                variant={"link"}
+                cursor={"pointer"}
+                minW={0}
+              >
+                <Avatar
+                  size={"sm"}
+                  src={"https://avatars.githubusercontent.com/u/38766452"}
+                />
+              </MenuButton>
+              <MenuList alignItems={"center"}>
+                <br />
+                <Center>
+                  <Avatar
+                    size={"2xl"}
+                    src={"https://avatars.githubusercontent.com/u/38766452"}
+                  />
+                </Center>
+                <br />
+                <Center>
+                  <p>{props?.session?.user?.name}</p>
+                </Center>
+                <br />
+                <MenuDivider />
+                <MenuItem as="a" href={Routes.DASHBOARD}>
+                  Your Organisations
+                </MenuItem>
+                <MenuItem as="a" href="/profile">
+                  Your Profile
+                </MenuItem>
+                <MenuDivider />
+                <MenuItem
+                  onClick={() => {
+                    signOut();
+                    navigateToAuthPage();
+                  }}
+                >
+                  Logout
+                </MenuItem>
+              </MenuList>
+            </Menu>
+          )}
+        </Stack>
+      </Flex>
+      <Collapse in={isOpen} animateOpacity>
+        <MobileNav />
+      </Collapse>
     </Box>
   );
 }
+
+const DesktopNav = () => {
```

Comment:
Quite a big file, I would recommend moving the different nav bars into own files